dali/test/python/test_operator_nonsilence.py

```diff
@@ -0,0 +1,103 @@
+                                     hop_length=hop_length)
+
+    # librosa's trim function calculates power with reference to center of window,
+    # while DALI uses beginning of window. Hence the subtraction below
+    begin = index[0] - frame_length / 2
+    length = index[1] - index[0]
+    return np.array([begin]), np.array([length])
+
+
+class NonsilencePipeline(Pipeline):
+    def __init__(self, batch_size, num_threads=1, exec_async=True, exec_pipelined=True):
+        super(NonsilencePipeline, self).__init__(batch_size, num_threads, 0, seed=42,
+                                                 exec_async=exec_async,
+                                                 exec_pipelined=exec_pipelined)
+        self.input = ops.FileReader(device="cpu", file_root=audio_files)
+        self.decode = ops.AudioDecoder(device="cpu", dtype=types.FLOAT, downmix=True)
+
+        self.nonsilence = None
+
+    def define_graph(self):
+        if self.nonsilence is None:
+            raise RuntimeError(
+                "Error: you need to derive from this class and define `self.nonsilence` operator")
+        read, _ = self.input()
+        audio, rate = self.decode(read)
+        begin, len = self.nonsilence(audio)
+        return begin, len
+
+
+class NonsilenceDaliPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_power,
+                 reset_interval):
+        super(NonsilenceDaliPipeline, self).__init__(batch_size, num_threads=1)
+        self.nonsilence = ops.NonsilentRegion(cutoff_db=cutoff_value, window_length=window_size,
+                                              reference_power=reference_power,
+                                              reset_interval=reset_interval,
+                                              device="cpu")
+
+
+class NonsilenceRosaPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_power, reset_interval):
+        super(NonsilenceRosaPipeline, self).__init__(batch_size, num_threads=1,
+                                                     exec_async=False, exec_pipelined=False)
+        hop_length = 1
+        function = partial(trim_ref, cutoff_value,
+                           np.max if reference_power == 0. else reference_power,
+                           window_size, hop_length)
+        self.nonsilence = ops.PythonFunction(function=function, num_outputs=2)
+
+
+def check_nonsilence_operator(batch_size, cutoff_value, window_size, reference_db,
+                              reset_interval, eps):
+    test_utils.compare_pipelines(
+        NonsilenceDaliPipeline(batch_size, cutoff_value, window_size, reference_db, reset_interval),
+        NonsilenceRosaPipeline(batch_size, -cutoff_value, window_size, reference_db,
+                               reset_interval), batch_size=batch_size, N_iterations=1, eps=eps)
+
+
+def test_nonsilence_operator():
+    batch_size = 1
+    eps = 1
```

Comment:
eps should be 0, or close to 0, shouldn't it?

Reply:
`eps` in this case measures integral values. 0 might be too narrow, since there might be numeric and rounding differences, so I went with 1